src/script/lua_api/l_util.cpp

```diff
@@ -320,6 +321,103 @@ int ModApiUtil::l_decompress(lua_State *L)
 	return 1;
 }
 
+#if USE_CURL
+void ModApiUtil::read_http_fetch_request(lua_State *L, HTTPFetchRequest &req)
+{
+	luaL_checktype(L, 1, LUA_TTABLE);
+
+	req.caller = httpfetch_caller_alloc();
+	getstringfield(L, 1, "url", req.url);
+	req.post_data = getstringfield_default(L, 1, "post_data", "");
+	std::string useragent = getstringfield_default(L, 1, "useragent", "");
+	lua_getfield(L, 1, "useragent");
+	bool useragent_exists = lua_isstring(L, 2);
+	lua_pop(L, 1);
+	if (useragent_exists)
+		req.useragent = getstringfield_default(L, 1, "useragent", "");
```

Comment:
Actually, minetest uses a builtin useragent, consisting of:
```C++
useragent = std::string("Minetest/") + minetest_version_hash + " (" + porting::get_sysinfo() + ")";
```
That is why I do the checking for the existance of that string. You might just want to use the default user-agent or you might want to hide the user agent completely, so use "".